### PR TITLE
Remove install file and harden sytemd unit a bit

### PR DESCRIPTION
**Pull request by brunnre8**
This removes the install file and replaces it with the systemd-sysuser / tmpfiles factories.

Further this hopefully splits the configs from the "data" directories by symlinking it like you suggested in IRC (I am bookworm)

Could you please review and tell me what you'd like to see changed?

Sorry for the delay, took a while due to $life

### CODE REVIEW

**Comment by xPaw** [comment]
I don't quite like the idea of using symlinks, as that diverses from all the other packaging methods Lounge has, and would only cause more harm than good (considering its symlinked, not actually moved).

File: .SRCINFO [hunk 1]
```diff
@@ -13,9 +12,13 @@ pkgbase = thelounge
 	source = http://registry.npmjs.org/thelounge/-/thelounge-2.7.1.tgz
 	source = system.service
 	source = user.service
-	sha256sums = 4b4970caba850042244798008afd334dd2ddd7e25339c864837ce11ba4703021
-	sha256sums = SKIP
-	sha256sums = SKIP
+	source = sysusers.d
+	source = tmpfiles.d
+	md5sums = f2be61e721cc677fb624687cf1e8944c
```
Comment (xPaw) [inline .SRCINFO]: Can you revert to using `sha256sums`?

File: PKGBUILD [hunk 2]
```diff
@@ -1,46 +1,48 @@
...
-	"https://registry.npmjs.org/$_npmname/-/$_npmname-$pkgver.tgz"
-	"system.service"
-	"user.service"
-)
-noextract=("$_npmname-$pkgver.tgz")
-sha256sums=(
-	'4b4970caba850042244798008afd334dd2ddd7e25339c864837ce11ba4703021'
-	'SKIP'
-	'SKIP'
+    "http://registry.npmjs.org/$pkgname/-/$pkgname-$pkgver.tgz"
```
Comment (xPaw) [inline PKGBUILD]: https

**Comment by xPaw** [comment]
Have you tried installing thelounge@next with this? Does it correctly handle installing/downloading sqlite3?

File: .SRCINFO (hunk 1 above)
Comment (brunnre8) [reply .SRCINFO]: yes, will do

File: PKGBUILD (hunk 2 above)
Comment (brunnre8) [reply PKGBUILD]: sorry, oversight

**Comment by brunnre8** [comment]
@xPaw the split config / data dirs was a suggestion via IRC.

Can you discuss between the maintainers what you prefer?

And no I didn't try next as the current npm release was broken due to a broken dependency (event-stream) when I tried

**Comment by xPaw** [comment]
> Can you discuss between the maintainers what you prefer?

Our general preference is to have minimal differences between packages, so just using /etc without symlinks would be preferable. Maybe in a future major release we can change it.

**Comment by brunnre8** [comment]
Fair, will change it.

@xPaw do you plan a new release on npm or is the build failure pebcak on my side?

**Comment by xPaw** [comment]
> @xPaw do you plan a new release on npm or is the build failure pebcak on my side?

It's certainly broken via yarn if you try to pull dev dependencies, however I'm not sure why you're running into that with npm, can you try with `--production` flag? You might also need `--unsafe-perm` to get sqlite to work.

**Comment by brunnre8** [comment]
I tried with yarn during testing, run into the issue and then didn't try npm as well.

I'll fix it up according to your suggestions

**Comment by brunnre8** [comment]
Addressed the review comments.

@xPaw rc5 builds fine with the pkgbuild via the npm registry, but I run Arch on x86 so that should be fairly straight forward.

**Review by xPaw — APPROVED** [review]

**Review by astorije — APPROVED** [review]

**Comment by maxpoulin64** [comment]
Looks good to me and verified working! Merging in a moment

**Comment by Superpiffer** [comment]
Forse some reason, user.service file has a wrong sha256sum:
```
sha256sum user.service
c609f3309f54bd6285e99ff29ca2464828bec7bbbca67243ee688bd2d605dbf0  user.service
```

**Comment by brunnre8** [comment]
Ack, the aur version doesn't have the updated version

@maxpoulin64  as I don't have push access yet, could you please have a look?
The git repo works fine, just the aur is broken

**Comment by maxpoulin64** [comment]
That's odd, it worked for me and they should both be the same master branch.

**Comment by maxpoulin64** [comment]
Seems to have been because master had #7 merged in so the checksums had changed between what was in this PR and what ended up in master.

**Pull request merged**